.clang-format

```diff
@@ -0,0 +1,40 @@
+BasedOnStyle: LLVM
```

Comment:
Can we rename this `.clang-format` to something else?  `example_clang_format`?  Just `clang_format`?

I love having an example `.clang-format` around that captures the basics of libMesh formatting guidelines, but (for MOOSE use) I also have a `.vimrc` that will helpfully notice that there's a `.clang-format` in an ancestor directory of any file I'm trying to save and will automatically run `clang-format` accordingly.